client/src/pages/CalendarPage/CalendarPage.js

```diff
@@ -116,17 +119,32 @@ const CalendarPage = () => {
     });
   };
 
-  const getTaskReflection = async (id) => {
-    await instance
-      .get(`/feedback/tasks/${id}`)
-      .then((taskReflectionData) => {
-        return taskReflectionData.data;
-      })
-      .catch(() => {
-        // this value doesn't matter since no task cards will be shown if
-        // at least one get task reflection fails
-        return { body: "", satisfaction: 0 };
-      });
+  const getTaskReflections = async (tasks) => {
```

Comment:
It's probably better if we can fetch the Task reflection just for a particular task rather than for all of them because the user might not be clicking on every single feedback. So it should be that upon a click on the "View" button, the feedback should be fetched then.

Reply:
Implemented as suggested. `getTaskReflection` is defined on `CalendarPage` and passed to `TaskCard` where `actionBtn` can call it as appropriate (i.e., when the "View" button is clicked).